### ISSUE TITLE
upgrade debug@4 dependency

### ISSUE THREAD

**Issue by hal869** [eslint/eslint#10854]
With

```
node -v
	v10.10.0
```

and,
```
├─ debug@4.0.0
├─ eslint@5.5.0
```

installed, where the debug 4 release is new/current,

https://github.com/visionmedia/debug/commit/7fb104b8cfcbc3a91d8e4a6727638c3fe24be8d2

this warning now appears on general package upgrades
```
warning eslint > debug@3.2.4: `debug` versions 3.2.0-3.2.2 break Node 4. Users of node LTS and onward (modern Node versions): please upgrade to `debug@4` or above as soon as possible. Node 4 users: please pin to `debug@>=3.2.3 <4` as it will be the last version range that supports Node@4.
```
